### PR TITLE
1 create redis terraform module

### LINKED ISSUE #1: Create Redis Terraform module

### PR DESCRIPTION
**Pull request by jakub-kubera**
## What changes does your PR bring?

In a few words, describe what changes your commits make to the code.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (non-breaking change to the build process or auxiliary tools)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (Improvements or additions to documentation)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## References

Fixes:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by michaltomczuk** [comment]
Readme file is missing in the example

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (stawii) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Don't change template in PR.

File: README.md [hunk 1]
```diff
@@ -15,12 +15,15 @@ What does the module provide?
 ## Usage
 
 ```hcl
-module "module_name" {
-  source  = "github.com/opsd-io/module_name?ref=v0.0.1"
-
-  # Variables
-  variable_1 = "foo"
-  variable_2 = "bar"
+module "terraform-module-digitalocean-redis" {
```
Comment (stawii) [inline README.md]: Simplify name, "do_redis", "redis_cluster" or generic "example". And please use `_`  not `-`.

Comment (stawii) [reply README.md]: The `source` is missing.

File: examples/create_redis_database/terraform.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">= 1.5.5"
+  required_providers {
```
Comment (stawii) [inline examples/create_redis_database/terraform.tf:4]: RTFM. Separate block from arguments with single empty line.

File: main.tf [hunk 3]
```diff
@@ -1 +1,26 @@
-# Terraform code goes here
+resource "digitalocean_database_redis_config" "main" {
+  cluster_id             = digitalocean_database_cluster.main.id
+  maxmemory_policy       = "allkeys-lru"
+  notify_keyspace_events = "KEA"
+  timeout                = 90
+}
+resource "digitalocean_database_firewall" "main" {
```
Comment (stawii) [inline main.tf:20]: RTFM. Separate resources with single empty line.

File: main.tf [hunk 4]
```diff
@@ -1 +1,26 @@
...
+
+  dynamic "rule" {
+    for_each = var.firewall_rules
+    content {
+      type  = "ip_addr"
+      value = rule.value
+    }
+  }
+}
+resource "digitalocean_database_cluster" "main" {
```
Comment (stawii) [inline main.tf:33]: RTFM. Separate resources with single empty line.

File: outputs.tf [hunk 5]
```diff
@@ -1,4 +1,52 @@
-# output "variable" {
-#   description = "output variable description"
-#   value       = variable.main.name
-# }
+output "main_host_id" {
```
Comment (stawii) [inline outputs.tf]: I'm not sure if `main_` and `redis_` prefixes are needed in outputs. Please rethink this.

File: versions.tf
Comment (stawii) [inline versions.tf:1]: RTFM. The terraform block in modules should be at beginning of the main.tf file.
Don't forget about empty line before `required_providers` block ;)

**Review by stawii — CHANGES_REQUESTED** [review]

**Review by jakub-kubera — COMMENTED** [review]
applied changes

**Review by stawii — APPROVED** [review]

**Pull request merged**